### PR TITLE
Fixing the gen_attention_mask_in_length function to handle the case when sequence id is -1 due to attention masking

### PR DESCRIPTION
**Pull request by ShashankMosaicML**

### CODE REVIEW

**Review by rajammanabrolu — APPROVED** [review]
lgtm thanks for fixing this so quickly!

**Review by dakinggg — APPROVED** [review]
LGTM
